pymc_experimental/model/marginal_model.py

```diff
@@ -248,14 +248,20 @@ def marginalize(self, rvs_to_marginalize: Union[TensorVariable, Sequence[TensorV
         # Raise errors and warnings immediately
         self.clone()._marginalize(user_warnings=True)
 
-    def unmarginalize(
+    def unmarginalize(self, rvs_to_unmarginalize):
+        for r in rvs_to_unmarginalize:
+            self.register_rv(rv, name=rv.name)
+
+        self.marginalized_rvs = list(set(self.marginalized_rvs) - set(rvs_to_unmargalize))
```

Comment:
Slight preference because it won't alter the order of the remaining marginalized_rvs. You had a typo though: for `r` instead of for `rv`

```suggestion
        for rv in rvs_to_unmarginalize:
            rv = self.marginalized_rvs.pop(rv)
            self.register_rv(rv, name=rv.name)
```